src/blocks/map/edit.js

```diff
@@ -31,65 +31,60 @@ import {
 	Icon,
 	TextControl,
 } from '@wordpress/components';
-import { useState, useEffect } from '@wordpress/element';
+import { Fragment, Component } from '@wordpress/element';
 import apiFetch from '@wordpress/api-fetch';
 
-const Edit = ( props ) => {
-	const {
-		attributes,
-		setAttributes,
-		isSelected,
-		className,
-	} = props;
+class Edit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			apiKey: '',
+			address: this.props.attributes.address,
+			coords: null,
+			hasError: false,
+			isSavedKey: false,
+			keySaved: false,
+		};
```

Comment:
For posterity: This change was needed because the previous refactor to hooks was incompatible with the use of API Keys for Google Maps. Reverting to the class based structure allows the Maps block to function as expected with or without the API Key.